Docs/reference/content/reference/driver/crud/client_side_encryption.md

```diff
@@ -181,4 +181,153 @@ namespace MongoDB.Driver.Examples
+### Explicit Encryption and Decryption
+
+Explicit encryption and decryption is a **MongoDB community** feature and does not use the `mongocryptd` process. Explicit encryption is provided by the `ClientEncryption` class. The following example has been adapted from [`ExplicitEncryptionExamples.cs`](https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Examples/ExplicitEncryptionExamples.cs):
+
+```
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using MongoDB.Driver.Encryption;
+using MongoDB.Libmongocrypt;
+
+namespace MongoDB.Driver.Examples
+{
+    public class ExplicitEncryptionExamples
+    {
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+
+        public static void Main(string[] args)
+        {
+            var localMasterKey = Convert.FromBase64String(LocalMasterKey);
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+            var localKey = new Dictionary<string, object>
+            {
+                { "key", localMasterKey }
+            };
+            kmsProviders.Add("local", localKey);
+
+            var keyVaultNamespace = CollectionNamespace.FromFullName("admin.datakeys");
+            var keyVaultClient = new MongoClient();
+            var keyVaultDatabase = keyVaultClient.GetDatabase(keyVaultNamespace.DatabaseNamespace.DatabaseName);
+            keyVaultDatabase.DropCollection(keyVaultNamespace.CollectionName);
+
+            // Create the ClientEncryption instance
+            var clientEncryptionSettings = new ClientEncryptionOptions(
+                keyVaultClient,
+                keyVaultNamespace,
+                kmsProviders);
+            var clientEncryption = new ClientEncryption(clientEncryptionSettings);
+
+            var dataKeyId = clientEncryption.CreateDataKey(
+                "local",
+                new DataKeyOptions(),
+                CancellationToken.None);
+
+            var originalString = "123456789";
+            Console.WriteLine($"Original string {originalString}.");
+
+            // Explicitly encrypt a field
+            var encryptOptions = new EncryptOptions(
+                EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic.ToString(),
+                keyId: dataKeyId);
+            var encryptedFieldValue = clientEncryption.Encrypt(
+                originalString,
+                encryptOptions,
+                CancellationToken.None);
+            Console.WriteLine($"Encrypted value {encryptedFieldValue}.");
+
+            // Explicitly decrypt the field
+            var decryptedValue = clientEncryption.Decrypt(encryptedFieldValue, CancellationToken.None);
+            Console.WriteLine($"Decrypted value {decryptedValue}.");
+
+            clientEncryption.Dispose();
```

Comment:
using statement instead of Dispose?

Reply:
It's a good idea to be consistent. I don't disagree with that.

Remember these are examples that show the user HOW to do something. If the example uses a `using` statement then the user probably will also. If the example does not use a `using` statement then the user probably won't either.

It's important that our examples be as correct as possible.

As currently coded if an exception is thrown before line 247 is reached then the `clientEncryption` object will NOT be Disposed. That's why a `using` statement is better.

Reply:
But at the same time, we need to show a user how to use our driver, but not teach him how he should write code (your suggestion looks is out of scope for this ticket).
However, I don't see a lot of harm to change it, so I will update it in both places

Reply:
done

Reply:
> But at the same time, we need to show a user how to use our driver, but not teach him how he should write code

I was not suggesting we need to teach them how to write code. Just that OUR code follow best practice.